src/test/java/core/basesyntax/strategy/PurchaseActionTest.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.strategy;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.actions.BalanceActionHandler;
+import core.basesyntax.strategy.actions.PurchaseActionHandler;
+import org.junit.Test;
+
+public class PurchaseActionTest {
+    @Test
+    public void purchaseAction_ok() {
+        Storage.getFruits().clear();
+        new BalanceActionHandler().apply(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                        "banana",
+                        10));
+        new PurchaseActionHandler().apply(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                        "banana",
+                        5));
```

Comment:
the same issue with initialization. Also, you should test only PurchaseActionHandler here. Think how you can avoid using BalanceActionHandler here. Fix similar issue in all places